From 19e0ae73e19fc3e545f268f7178aa69ea9ea5b72 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Wiktor=20Strz=C4=99ba=C5=82a?= <wiktorek140@gmail.com>
Date: Mon, 13 Jul 2020 20:13:25 +0200
Subject: [PATCH 137/210] arm64: dts: qcom: msm8916-motorola-surnia: add
 display (v4)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Wiktor Strzębała <wiktorek140@gmail.com>

v2: Make reset-gpios GPIO_ACTIVE_LOW
v3: Use generic motorola,surnia-panel compatible
v4: Add &mdss { status = "okay"; };
Signed-off-by: Stephan Gerhold <stephan@gerhold.net>
---
 .../boot/dts/qcom/msm8916-motorola-surnia.dts | 50 +++++++++++++++++++
 1 file changed, 50 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-motorola-surnia.dts b/arch/arm64/boot/dts/qcom/msm8916-motorola-surnia.dts
index cd919e4b0e1..c08713a8e25 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-motorola-surnia.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-motorola-surnia.dts
@@ -102,6 +102,39 @@ &blsp1_uart1 {
 	status = "okay";
 };
 
+&dsi0 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&mdss_default>;
+	pinctrl-1 = <&mdss_sleep>;
+
+	panel@0 {
+		compatible = "motorola,surnia-panel";
+		reg = <0>;
+
+		power-supply = <&pm8916_l17>;
+		reset-gpios = <&msmgpio 78 GPIO_ACTIVE_LOW>;
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&dsi0_out>;
+			};
+		};
+	};
+};
+
+&dsi0_out {
+	data-lanes = <0 1>;
+	remote-endpoint = <&panel_in>;
+};
+
+&dsi_phy0 {
+	qcom,dsi-phy-regulator-ldo-mode;
+};
+
+&mdss {
+	status = "okay";
+};
+
 &pm8916_resin {
 	status = "okay";
 	linux,code = <KEY_VOLUMEDOWN>;
@@ -270,6 +303,23 @@ gpio_keys_default: gpio-keys-default {
 		bias-pull-up;
 	};
 
+	mdss {
+		mdss_default: mdss-default {
+			pins = "gpio78";
+			function = "gpio";
+
+			drive-strength = <8>;
+			bias-disable;
+		};
+		mdss_sleep: mdss-sleep {
+			pins = "gpio78";
+			function = "gpio";
+
+			drive-strength = <2>;
+			bias-pull-down;
+		};
+	};
+
 	ts_int_default: ts-int-default {
 		pins = "gpio21";
 		function = "gpio";
-- 
2.34.1

